ResearchKit/Common/ORKFormStep.h

```diff
@@ -129,6 +129,14 @@ ORK_CLASS_AVAILABLE
 @property (nonatomic, copy, readonly) NSString *identifier;
 
 /**
+ A Boolean value indicating whether the form item is optional.
+ 
+ The default value of this property is `NO`. When the value is `YES`, this form item doesn't need
```

Comment:
That's almost right.

Basically, the `optional` setting in a *form step* only controls if the *Skip button* shows or not (skipping a form step nullifies all previously provided answers, so it's a slightly different use case).

Regardless of the *form step* optionality, then we have:

FormItems  | *Done button* enabled
------------- | -------------
All non-optional  | If all items are answered
Mixed  | If all non-optional items are answered
All optional | If at least one item is answered

Tapping the *Done button* does not nullify any of the provided answers.